### PR TITLE
HSEARCH-1933 Warnings during build

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/HSEARCH-1933

### CODE REVIEW

File: backends/jms/pom.xml [hunk 1]
```diff
@@ -41,6 +41,12 @@
             <type>test-jar</type>
             <scope>test</scope>
         </dependency>
+
+        <!-- Avoid warnings during build -->
+        <dependency>
```
Comment (Sanne) [inline backends/jms/pom.xml]: but we don't want to have this as a transitive dependency?

Comment (DavideD) [inline backends/jms/pom.xml]: It's provided, as it is defined as provided in the dependency management of the aggregator

Comment (DavideD) [inline backends/jms/pom.xml]: I can add the scope to make it explicit

Comment (Sanne) [inline backends/jms/pom.xml]: ok but is "provided" correct? That maven definition always confuses me, but we don't really expect it to be provided at runtime.

Comment (DavideD) [inline backends/jms/pom.xml]: To be precise, it's provided and optional: https://github.com/hibernate/hibernate-search/blob/master/pom.xml#L549
I don't think there is a better scope in maven that match what we need for our use case.

Comment (Sanne) [inline backends/jms/pom.xml]: ok

**Comment by Sanne** [comment]
rebased and merged

**Pull request closed without merging**